node_modules/oae-transfer/transfer/js/transfer.js

```diff
@@ -0,0 +1,291 @@
+                    return false;
+                });
+            });
+
+            // When the user valide the for do make a transfer
+            $rootel.on('click', '#transfer-makeTransfer-btn', function(e) {
+                // Avoid the reload
+                e.preventDefault();
+                makeTransfer(function(err){
+                    // To be sure that we wait the insert of transfer before trying to make a get on it
+                    setTimeout(function(){
+                        closeModal();
+                    }, 100);
+                    // Avoid default form submit behavior
+                    return false;
+                });
+            });
+
+            // When the user click on cancel transfer
+            $rootel.on('click', '#transfer-code-cancel-transfer-btn', function() {
+                deleteTransfer(function(err){
+                    // Avoid default form submit behavior
+                    return false;
+                });
+            });
+        };
+
+        // Access to API
+
+        /**
+        * When the user click on validate in the generate code panel
+        */
+        var createTransfer = function(callback){
+            // Get alls necessary data            
+            var emailOrigin = profile.email;
+            var idUserOrigin = profile.id;
+            var emailTarget = $.trim($('#user-autosuggest', $rootel).val());
+
+            if(emailTarget === emailOrigin){
+                oae.api.util.notification(
+                    oae.api.i18n.translate('__MSG__TRANSFER_CREATION_FAILED__'),
+                    oae.api.i18n.translate('__MSG__YOUR_TRANSFER_IS_NOT_CREATED_SAME_EMAIL__', 'preferences'),
+                    'error'
+                );
+                return callback(null);
+            } else { 
+                oae.api.transfer.createTransfer(emailOrigin, emailTarget, idUserOrigin, function(err, data){
+                    if(!err){
+                        // Show a success notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__TRANSFER_CREATED__', 'preferences'),
+                            oae.api.i18n.translate('__MSG__YOUR_TRANSFER_IS_CREATED__', 'preferences')
+                        );
+                    } else {
+                        // Show a generic failure notification
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__TRANSFER_CREATION_FAILED__'),
+                            oae.api.i18n.translate('__MSG__YOUR_TRANSFER_IS_NOT_CREATED__', 'preferences'),
+                            'error'
+                        );
+                    }
+                }); 
+                return callback(null);
+            }
+        }
+
+        /**
+        * Verify the user have already start a procedure of transfer
+        */
+        var isTransferAlreadyCreated = function(){
```

Comment:
Important note:

It seems that you're assuming that there is one transfer per e-mail account, but there can be several accounts with the same e-mail address. Let's take the following scenario:

- There is user kermit on tenancy cambridge.oae.com, using kermit@puppets.com
- The same user kermit is also on tenancy oxford.oae.com, using the same e-mail

Right now, if the user kermit @oxford.oae.com generates a transfer code, it will appear on both accounts! And if one activated the code, which account will it import??

That means that the transfer code must be associated with a specific account, not the e-mail. Having it like this can lead to unexpected behaviour.

Other comments:

- it doesn't make much sense to have a method starting `isTransferAlreadyCreated` and then returning nothing. I was expecting it to return true or false, as that is common practice.
- You're missing some spaces between `) {` all around